### PR TITLE
deprecate usage of :index_type => :document_type

### LINKED ISSUE #102: Parameter "index_type" should resemble Elasticsearch lingo (more closely)
Consider me a user. **)*
I had a hard (and embarrassing) time figuring out the parameter for setting the `doc_type` (used in python) or `_type` (native) for the Elasticsearch output. Here I'll take the embarrassment one level further and blame it on you ;)

It may seem obvious once learned, but: coming from Elasticsearch, I *never ever* encountered the term `index_type`. Please consider changing or aliasing the parameter name, or at least mention it in the docs for `index_type` with one of the more typical synonyms. Maybe use `document_type`, since you already use `document_id`.

Furthermore I wonder, if the docs are correct?
```
index_type
*    Value type is string
*    There is no default value for this setting. 
```
it may have happened somewhere else in the processing chain, but without setting the parameter, my events ended up with the mapping-type `logs`.

*) well, maybe a little absent-minded, but that's how some of us are...

### PR DESCRIPTION
**Pull request by talevy**
Some users feel this jargin does not match nicely
with Elasticsearch. So a renaming is warranted.

Closed #102.

### CODE REVIEW

File: lib/logstash/outputs/elasticsearch.rb [hunk 1]
```diff
@@ -49,7 +49,13 @@ class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
 
   # The index type to write events to. Generally you should try to write only
   # similar events to the same 'type'. String expansion `%{foo}` works here.
-  config :index_type, :validate => :string
+  # 
+  # Depricated in favor of `document_type` field.
```
Comment (suyograo) [inline lib/logstash/outputs/elasticsearch.rb]: typo.. Deprecated

**Comment by suyograo** [comment]
Small typo, otherwise LGTM

File: lib/logstash/outputs/elasticsearch.rb (hunk 1 above)
Comment (talevy) [inline lib/logstash/outputs/elasticsearch.rb]: done.

**Comment by suyograo** [comment]
:ship:

**Pull request closed without merging**